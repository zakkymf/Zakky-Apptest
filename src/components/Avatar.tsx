import React from 'react';
import {Image, ImageStyle, StyleProp, StyleSheet} from 'react-native';
import images from '../assets/images';

interface AvatarProps {
  source: string;
  avatarStyle?: StyleProp<ImageStyle>;
}

const Avatar: React.FC<AvatarProps> = ({source, avatarStyle}) => {
  return (
    <Image
      style={avatarStyle ?? styles.avatar}
      source={source !== 'N/A' ? {uri: source} : images.default_avatar}
    />
  );
};

const styles = StyleSheet.create({
  avatar: {
    width: 40,
    height: 40,
    borderRadius: 20,
  },
});

export default Avatar;
