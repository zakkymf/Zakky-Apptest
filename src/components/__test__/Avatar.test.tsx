import React from 'react';
import {render} from '@testing-library/react-native';
import Avatar from '../Avatar';

describe('@components/Avatar', () => {
  it('render avatar component', () => {
    const component = () =>
      render(<Avatar source="https://www.helpforenglish.cz/files/wiki.jpeg" />);

    const {toJSON} = component();
    expect(toJSON()).toMatchSnapshot();
  });

  it('render default avatar', () => {
    const component = () => render(<Avatar source="N/A" />);

    const {toJSON} = component();
    expect(toJSON()).toMatchSnapshot();
  });
});
