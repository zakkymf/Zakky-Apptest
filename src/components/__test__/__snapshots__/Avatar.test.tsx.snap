// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Avatar render avatar component 1`] = `
<Image
  source={
    Object {
      "uri": "https://www.helpforenglish.cz/files/wiki.jpeg",
    }
  }
  style={
    Object {
      "borderRadius": 20,
      "height": 40,
      "width": 40,
    }
  }
/>
`;

exports[`@components/Avatar render default avatar 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/images/default_avatar.png",
    }
  }
  style={
    Object {
      "borderRadius": 20,
      "height": 40,
      "width": 40,
    }
  }
/>
`;
